.github/workflows/common.yml

```diff
@@ -77,6 +83,9 @@ jobs:
         os: [ubuntu-latest]
         python-version: ["3.9", "3.10", "3.11"]
     timeout-minutes: 20
+    concurrency:
+      group: 'test:common:${{ github.ref }}'
```

Comment:
Consider making the concurrency group name more specific by including the job name and Python version to allow parallel runs of different jobs:
```yaml
group: 'test:${{ github.job }}:${{ matrix.python-version }}:${{ github.ref }}'
```
This would allow different Python versions to run in parallel while still cancelling redundant runs of the same version.